src/Mvc/Mvc.Core/src/DependencyInjection/MvcCoreServiceCollectionExtensions.cs

```diff
@@ -65,7 +65,18 @@ public static IMvcCoreBuilder AddMvcCore(this IServiceCollection services)
 
         private static void ConfigureDefaultFeatureProviders(ApplicationPartManager manager)
         {
-            if (!manager.FeatureProviders.OfType<ControllerFeatureProvider>().Any())
+            var controllerFeaturePresent = false;
+
+            foreach(var feature in manager.FeatureProviders)
+            {
+                if(feature.GetType() == typeof(ControllerFeatureProvider))
```

Comment:
This is also a behavioral change: OfType is the equivalent of `feature is ControllerFeatureProvider`, which is different from `feature.GetType() == typeof(ControllerFeatureProvider)`, in particular if the actual instance is null (in which case the new version will null ref) or if the actual instance is derived from ControllerFeatureProvider (in which case the previous version would find it and the new version wouldn't).

Reply:
Ohhhh. I see. Seems like I need to tweek if condition there. I will get that changed!